tests/Sdk/Unit/Trace/TracerProviderTest.php

```diff
@@ -21,4 +25,41 @@ public function gettingSameTracerMultipleTimesShouldReturnSameObject()
 
         self::assertSame($tracer1, $tracer2);
     }
+
+    /**
+     * @test
+     */
+    public function newTraceProviderDefaultsToAlwaysOnSampler()
+    {
+        $traceProvider = new TracerProvider();
+        $description = $traceProvider->getSampler()->getDescription();
+
+        self::assertSame($description, 'AlwaysOnSampler');
+    }
+
+    /**
+     * @test
+     */
+    public function newTraceProviderAcceptsOtherSamplers()
+    {
+        $traceProvider = new TracerProvider(null, new AlwaysOffSampler());
+        $description = $traceProvider->getSampler()->getDescription();
+
+        self::assertSame($description, 'AlwaysOffSampler');
+
+        $traceProvider = new TracerProvider(null, new ParentBased());
+        $description = $traceProvider->getSampler()->getDescription();
+
+        self::assertSame($description, 'ParentBased');
+
+        $traceProvider = new TracerProvider(null, new AlwaysOnSampler());
+        $description = $traceProvider->getSampler()->getDescription();
+
+        self::assertSame($description, 'AlwaysOnSampler');
+
+        $traceProvider = new TracerProvider(null, new TraceIdRatioBasedSampler(1.0));
```

Comment:
Maybe we should have this test be something other than `1.0` - for the assertion.  No strong feelings about this just a thought.

Reply:
I could change it to 0.5 or something like that.  The main thing was to check the samplers were getting used, since I thought the samplers' functionality had already been thoroughly unit tested.

Reply:
Yup; I think something like 0.5 makes sense to me - 1.0 might be easily confused with `AlwaysOn` at some point.